src/Wellcome.Dds/Wellcome.Dds.Dashboard/Controllers/DashController.cs

```diff
@@ -0,0 +1,840 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using DlcsWebClient.Config;
+using DlcsWebClient.Dlcs;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+using Utils;
+using Utils.Caching;
+using Utils.Logging;
+using Wellcome.Dds.AssetDomain;
+using Wellcome.Dds.AssetDomain.Dashboard;
+using Wellcome.Dds.AssetDomain.Dlcs;
+using Wellcome.Dds.AssetDomain.Dlcs.Ingest;
+using Wellcome.Dds.AssetDomain.Dlcs.Model;
+using Wellcome.Dds.AssetDomain.Workflow;
+using Wellcome.Dds.AssetDomainRepositories.Mets;
+using Wellcome.Dds.Common;
+using Wellcome.Dds.Dashboard.Models;
+
+namespace Wellcome.Dds.Dashboard.Controllers
+{
+    [Authorize]
+    public class DashController : Controller
+    {
+        private readonly IDashboardRepository dashboardRepository;
+        private readonly IIngestJobRegistry jobRegistry;
+        private readonly IIngestJobProcessor jobProcessor;
+        private readonly ISimpleCache cache;
+        private readonly IStatusProvider statusProvider;
+        private readonly IDatedIdentifierProvider recentlyDigitisedIdentifierProvider;
+        private readonly IWorkStorageFactory workStorageFactory;
+        private readonly CacheBuster cacheBuster;
+        private readonly DdsOptions ddsOptions;
+        private readonly DlcsOptions dlcsOptions;
+        private readonly IDds dds;
+        private readonly IWebHostEnvironment webHostEnvironment;
+        private readonly IWorkflowCallRepository workflowCallRepository;
+
+        const string CacheKeyPrefix = "dashcontroller_";
+        const int CacheSeconds = 5;
+        private Synchroniser synchroniser;
+
+        public DashController(
+            IDashboardRepository dashboardRepository,
+            ISimpleCache cache,
+            IIngestJobRegistry jobRegistry,
+            IIngestJobProcessor jobProcessor,
+            IStatusProvider statusProvider,
+            Synchroniser synchroniser,
+            IDatedIdentifierProvider recentlyDigitisedIdentifierProvider,
+            IWorkStorageFactory workStorageFactory,
+            CacheBuster cacheBuster,
+            IOptions<DdsOptions> ddsOptions,
+            IOptions<DlcsOptions> dlcsOptions,
+            IDds dds,
+            IWebHostEnvironment webHostEnvironment,
+            IWorkflowCallRepository workflowCallRepository)
```

Comment:
Not for now but just looking at dep list this is doing too much

Reply:
Yes agreed, this Controller is doing all the work and needs to be split up. Which is not quite the same problem but they are related.